src/sentry/llm/providers/preview.py

```diff
@@ -0,0 +1,22 @@
+from typing import Any
+
+from sentry.llm.providers.base import LlmModelBase
+
+
+class PreviewLLM(LlmModelBase):
```

Comment:
personally would try and avoid inheritance for this -- sentry historically has leaned on this a lot and it ends up being unmaintainably tangled (especially when attempting to type properly or reason about the implementation(s))

Reply:
i feel like in this particular care inheritance seems sensible, as the different providers should all conform to the same interface -- curious what you think an alternative implementation could look like